test/integration/http/http.go

```diff
@@ -0,0 +1,177 @@
+package http
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/test/cluster"
+	"gotest.tools/assert"
+
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/davecgh/go-spew/spew"
+	vegeta "github.com/tsenart/vegeta/v12/lib"
+)
+
+type HttpClusterTestRunner struct {
+	cluster.ClusterTestRunnerBase
+}
+
+func int32Ptr(i int32) *int32 { return &i }
+
+const minute time.Duration = 60
+
+var httpbinDep *appsv1.Deployment = &appsv1.Deployment{
+	TypeMeta: metav1.TypeMeta{
+		APIVersion: "apps/v1",
+		Kind:       "Deployment",
+	},
+	ObjectMeta: metav1.ObjectMeta{
+		Name: "httpbin",
+	},
+	Spec: appsv1.DeploymentSpec{
+		Replicas: int32Ptr(1),
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{"application": "httpbin"},
+		},
+		Template: apiv1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"application": "httpbin",
+				},
+			},
+			Spec: apiv1.PodSpec{
+				Containers: []apiv1.Container{
+					{
+						Name:            "httpbin",
+						Image:           "docker.io/kennethreitz/httpbin",
+						ImagePullPolicy: apiv1.PullIfNotPresent,
+						Ports: []apiv1.ContainerPort{
+							{
+								Name:          "http",
+								Protocol:      apiv1.ProtocolTCP,
+								ContainerPort: 80,
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+}
+
+func sendReceive(servAddr string) {
+	return
+}
+
+func (r *HttpClusterTestRunner) RunTests(ctx context.Context) {
+	//TODO https://github.com/skupperproject/skupper/issues/95
+	//all this hardcoded sleeps must be fixed, probably along with #95
+	//for now I am just keeping them in the same values that we are using
+	//for tcp_echo test, since in case of reducing test may fail
+	//intermitently
+
+	r.Pub1Cluster.GetService("httpbin", 3*minute)
+	time.Sleep(20 * time.Second) //TODO XXX What is the right condition to wait for?
+
+	r.Pub1Cluster.KubectlExecAsync(fmt.Sprintf("port-forward service/httpbin 8080:80"))
```

Comment:
As you commented in #95 would it be good to start with using client-go/tools/portforward from the get go?

Reply:
Correct, that is the ideal, emmm although it is not that straightforward, I am taking a look to that, if I find how to do it, will modify this part

Reply:
One small comment is that it might be interesting to use a dynamic local port, instead of hard coding it to use 8080.
Something like: `port-forward service/httpbin :80` (then we must parse kubectl output to get the dynamic port).

For upstream CI only this seems like not needed now (and eventually we can simply log an enhancement for that at
this point), but thinking of people trying to run it locally, this might be helpful as 8080 is used by default by several apps.